integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py

```diff
@@ -301,6 +439,86 @@ def write_documents(self, documents: List[Document], policy: DuplicatePolicy = D
+        self,
+        documents: List[Document],
+        policy: DuplicatePolicy = DuplicatePolicy.FAIL,
+    ) -> int:
+        """
+        Writes (or overwrites) documents into the store.
+
+        :param documents:
+            A list of documents to write into the document store.
+        :param policy:
+            Not supported at the moment.
+
+        :raises ValueError:
+            When input is not valid.
+
+        :returns:
+            The number of documents written
+        """
+        await self._ensure_initialized_async()
+        assert self._collection is not None
+
+        for doc in documents:
+            if not isinstance(doc, Document):
+                msg = "param 'documents' must contain a list of objects of type Document"
+                raise ValueError(msg)
+
+            if doc.content is None:
+                logger.warning(
+                    "ChromaDocumentStore cannot store documents with `content=None`. "
+                    "Document with id {doc_id} will be skipped.",
+                    doc_id=doc.id,
+                )
+                continue
+            elif hasattr(doc, "blob") and doc.blob is not None:
+                logger.warning(
+                    "Document with id {doc_id} contains the `blob` field. "
+                    "ChromaDocumentStore cannot store `blob` fields. "
+                    "This field will be ignored.",
+                    doc_id=doc.id,
+                )
+            data = {"ids": [doc.id], "documents": [doc.content]}
+
+            if doc.meta:
+                valid_meta = {}
+                discarded_keys = []
+
+                for k, v in doc.meta.items():
+                    if isinstance(v, SUPPORTED_TYPES_FOR_METADATA_VALUES):
+                        valid_meta[k] = v
+                    else:
+                        discarded_keys.append(k)
+
+                if discarded_keys:
+                    logger.warning(
+                        "Document {doc_id} contains `meta` values of unsupported types for the keys: {keys}. "
+                        "These items will be discarded. Supported types are: {types}.",
+                        doc_id=doc.id,
+                        keys=", ".join(discarded_keys),
+                        types=", ".join([t.__name__ for t in SUPPORTED_TYPES_FOR_METADATA_VALUES]),
+                    )
+
+                if valid_meta:
+                    data["metadatas"] = [valid_meta]
+
+            if doc.embedding is not None:
+                data["embeddings"] = [doc.embedding]
+
+            if hasattr(doc, "sparse_embedding") and doc.sparse_embedding is not None:
+                logger.warning(
+                    "Document {doc_id} has the `sparse_embedding` field set, "
+                    "but storing sparse embeddings in Chroma is not currently supported. "
+                    "The `sparse_embedding` field will be ignored.",
+                    doc_id=doc.id,
+                )
```

Comment:
to avoid code duplication, this part can become a sync method (something like `_prepare_data_for_chroma`) that we use in both `write_documents` and `write_documents_async`